src/tests/system/tests/test_cache.py

```diff
@@ -146,70 +146,174 @@ def test_cache__writes_to_both_database_files_when_using_fully_qualified_names(
+    client.sssd.start()
+
+    assert client.tools.getent.group("group1"), "Group group1 not found!"
+    assert client.tools.getent.group("group2"), "Group group2 not found!"
+
+    group1.add_member(provider.user("user1").add())
+    group2.delete()
+
+    client.sssctl.cache_expire(everything=True)
+
+    assert client.tools.getent.group("group1"), "Group group1 not found!"
+    assert "user1" in str(client.tools.getent.group("group1")), "Group group1 is missing a user!"
+    assert not client.tools.getent.group("group2"), "Group group2 found!"
+
+
+@pytest.mark.integration
+@pytest.mark.importance("low")
+@pytest.mark.parametrize("obj", ["user", "group", "netgroup"])
+@pytest.mark.parametrize("dbs", ["cache", "timestamps"])
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_cache__invalidate_entries_in_domain_and_timestamps_caches(
+    client: Client,
+    provider: LDAP,
+    obj: str,
+    dbs: str,
+):
+    """
+    :title: Invalidates object entries in the domain and timestamps caches using dataExpireTimestamp attribute
+    :setup:
+        1. Create user, group and netgroup
+        2. Add user as a member to group and netgroup
+        3. Start SSSD and lookup user, group and netgroup
+    :steps:
+        1. Search for object attribute dataExpireTimestamp in domain and timestamps ldb caches
+        2. Clear the SSSD cache for the object
+        3. Search for the object attribute dataExpireTimestamp in domain and timestamps ldb caches
+    :expectedresults:
+        1. Attribute is found
+        2. Cache is cleared for the object
+        3. Attribute is found and the value is '[1]' meaning it is cleared
+    :customerscenario: False
+    """
+    # Unable to parametrize provider method when adding objects, so all object types are created
+    user = provider.user("user").add()
+    provider.group("group").add().add_member(user)
+    provider.netgroup("netgroup").add().add_member(user=user)
+    client.sssd.start()
+
+    assert client.tools.getent.passwd("user")
+    assert client.tools.getent.group("group")
+    assert client.tools.getent.netgroup("netgroup")
+
+    path = f"/var/lib/sss/db/{dbs}_{client.sssd.default_domain}.ldb"
+
+    # netgroups does not use the fully qualified name and does not use the timestamp cache
+    if obj == "netgroup":
+        suffix = f"cn=netgroups,cn={client.sssd.default_domain},cn=sysdb"
+        ldb_filter = f"dn=name={obj},{suffix}"
+        if dbs == "timestamps":
+            pytest.skip("netgroup do not support timestamps database")
+    else:
+        suffix = f"cn={obj}s,cn={client.sssd.default_domain},cn=sysdb"
+        ldb_filter = f"dn=name={obj}@{client.sssd.default_domain},{suffix}"
+
+    result = client.ldb.search(path, suffix, filter=ldb_filter)
+    assert result != {}, f"ldbsearch {ldb_filter} did not return any results!"
+    for _, v in result.items():
+        result_expire_time = v.get("dataExpireTimestamp")
+        if result_expire_time is not None:
```

Comment:
Better assert it is not 'None'.

Reply:
Hrmm.... 

```
(Pdb) client.tools.getent.netgroup("netgroup")
netgroup (-,user,))

(Pdb) client.ldb.search(path, suffix)
{}

(Pdb) client.ldb.search(f"/var/lib/sss/db/cache_{client.sssd.default_domain}.ldb", suffix)
{'name=netgroup,cn=Netgroups,cn=test,cn=sysdb': {'dn': ['name=netgroup,cn=Netgroups,cn=test,cn=sysdb'], 'createTimestamp': ['1729646403'], 'name': ['netgroup'], 'objectClass': ['netgroup'], 'originalDN': ['cn=netgroup,ou=netgroups,dc=ldap,dc=test'], 'originalModifyTimestamp': ['20241023012002Z'], 'netgroupTriple': ['(-, user, )'], 'lastUpdate': ['1729646403'], 'dataExpireTimestamp': ['1729651803'], 'distinguishedName': ['name=netgroup,cn=Netgroups,cn=test,cn=sysdb']}}

(Pdb) client.ldb.search(f"/var/lib/sss/db/timestamps_{client.sssd.default_domain}.ldb", suffix)
{}

I checked it against the client system, and the object doesn't exist in the timestamps db. 
```

Reply:
I guess you replied to a wrong comment.
But this might be a real bug.